sycl/cmake/modules/AddSYCLDeviceTarget.cmake

```diff
@@ -0,0 +1,24 @@
+macro(add_sycl_executable ARG_TARGET_NAME)
+  cmake_parse_arguments(ARG
+    ""
+    "OPTIONS"
+    "SOURCES"
+    ${ARGN})
+
+  set(CXX_COMPILER clang++)
+  if(MSVC)
+      set(CXX_COMPILER clang-cl.exe)
+  endif()
+  set(DEVICE_COMPILER_EXECUTABLE ${LLVM_RUNTIME_OUTPUT_INTDIR}/${CXX_COMPILER})
+
+  add_custom_target(${ARG_TARGET_NAME}_exec ALL
+    COMMAND ${DEVICE_COMPILER_EXECUTABLE} -fsycl ${ARG_OPTIONS} ${ARG_SOURCES}
+      -o ${CMAKE_CURRENT_BINARY_DIR}/${ARG_TARGET_NAME}
+    BYPRODUCTS ${CMAKE_CURRENT_BINARY_DIR}/${ARG_TARGET_NAME}
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
+  add_dependencies(${ARG_TARGET_NAME}_exec sycl clang)
```

Comment:
Why do you need a dependency on clang and not clang++?
Also, if you use clang you may employ driver-mode=g++ command line argument.

Reply:
`clang++` is only a symlink to `clang`. I believe this should be fine.

Reply:
> `clang++` is only a symlink to `clang`. I believe this should be fine.

What about Windows?

Reply:
Fair point. I'll double check this on windows.

Reply:
I think you are fine. If I understand this line correctly, here we add dependency on cmake target, not executable name and AFAIK, clang cmake target builds all drivers: clang, clang++ and clang-cl.

Reply:
@alexbatashev, @s-kanaev, is this resolved?